### ISSUE TITLE
Metric: Precision-Recall, Density-Coverage

### ISSUE THREAD

**Issue by hecoding** [photosynthesis-team/piq#240]
**Is your feature request related to a problem? Please describe.**
Hi, I saw you have some common feature-based metrics working nicely on GPU. I was wondering if you would be interested in including a couple metrics that don't comprise the whole image assessment into a single scalar, like FID or KID do. But rather having separated quality and diversity scores to distinguish different failure cases, i.e. realistic but low diversity, low-quality but high diversity.

This metrics are Improved Precision and Recall [(P&R)](https://arxiv.org/abs/1904.06991) and Density and Coverage [(D&C)](https://arxiv.org/abs/2002.09797)

**Describe the solution you'd like**
I found released numpy/sklearn [code](https://github.com/clovaai/generative-evaluation-prdc/blob/master/prdc/prdc.py) that could be kinda straight-away ported to pytorch.

Cheers.

**Comment by zakajd** [comment]
Hi @hecoding 
Thanks for opening an issue.
Currently all core maintainers are busy, but we would gladly accept PR with the proposed metrics implemented. 
Please follow our [contribution guide ](https://github.com/photosynthesis-team/piq/blob/master/CONTRIBUTING.rst) and feel free to ask any questions 

Best regards

**Comment by hecoding** [comment]
Cool, will do. 
Yeah, the issue was for checking if a PR on these two was welcomed. I'll allocate some time then.
Thanks for the answer.

**Comment by hecoding** [comment]
Hi @zakajd
The first PR is done. Second will be Density and Coverage.
Can you review / assign somebody to review?
